mod/profile/classes/ElggPlugin/Profile/AnnotationMigration.php

```diff
@@ -21,7 +22,22 @@ public function getVersion() {
 	 * {@inheritdoc}
 	 */
 	public function shouldBeSkipped() {
-		return false;
+		// @todo We need somehow restrict this upgrade to users created before upgrade 3.x was performed
```

Comment:
can't we set a 'last_upgraded_ts' in the config? update this ts after a full run of upgrade(.php) and set to 0 at the end of installer ... we can then just check against accounts created before `last_upgrade_ts`... something like that

Reply:
I think a check for existing upgrades should be enough for now

Reply:
i thought that check was already part of the upgrades system itself (because with return false here, it still dissapears after a run)

besides, it won't fix the issue with this upgrade. Could you remove that from the description?

Reply:
Part mf me wants to allow upgrades to run again based on shouldBeSkipped().